packages/aws_common/lib/src/logging/aws_logger.dart

```diff
@@ -63,6 +65,22 @@ class AWSLogger implements Closeable {
 
   final Logger _logger;
 
+  /// Parent of this logger in the logger hierarchy.
+  AWSLogger? get _parent {
+    return activeLoggers[_logger.parent?.fullName];
+  }
+
+  /// Children of this logger in the logger hierarchy.
+  final List<AWSLogger> _children = [];
+
+  Never get _pluginAlreadyRegistered {
+    final loggerInstance = '$runtimeTypeName($namespace)';
+    throw StateError(
+        'A plugin with same type is already registered to "$loggerInstance"'
+        ' in the same logging hierarchy. Unregister the existing plugin from'
+        ' "$loggerInstance" first and then register the new plugin.');
```

Comment:
Typically would add a trailing comma here - not sure why the analyzer didn't catch it